sklearn/linear_model/_logistic.py

```diff
@@ -1042,10 +1042,12 @@ class LogisticRegression(LinearClassifierMixin,
     Parameters
     ----------
     penalty : {'l1', 'l2', 'elasticnet', 'none'}, default='l2'
-        Used to specify the norm used in the penalization. The 'newton-cg',
-        'sag' and 'lbfgs' solvers support only l2 penalties. 'elasticnet' is
-        only supported by the 'saga' solver. If 'none' (not supported by the
-        liblinear solver), no regularization is applied.
+        Used to specify the norm used in the penalization.
+        If 'none' (not supported by the liblinear solver),
+        no regularization is applied.
+
+        Note: Some of penalties may not work with specific solver.
+        See parameter 'solver' below.
```

Comment:
```suggestion
        Specify the norm of the penalty:

        - `'none'`: no penalty is added;
        - `'l2'`: add a L2 penalty term and it is the default choice;
        - `'l1'`: add a L1 penalty term;
        - `'elasticnet'`: both L1 and L2 penalty terms are added.

        .. note::
           Some penalties may not work with specific solver. See parameter
           `solver` below, to know the compatibility between the penalty and
           solver.
```

Reply:
Let's add a bit more information regarding the penalties (elasticnet might not be a given for everyone) and use the a sphinx directive to highlight the note.

Reply:
Good Point! 
About LogisticRegressionCV I wasn't quite sure if update that too, but I will do that